lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/DBOGroupMembersDAOImpl.java

```diff
@@ -81,7 +86,9 @@ public void addMembers(String groupId, List<String> memberIds)
 		List<String> locks = new ArrayList<String>(memberIds);
 		locks.add(groupId);
 		for (Long id : sortIds(locks)) {
-			userGroupDAO.getEtagForUpdate(id.toString());
+			String etag = userGroupDAO.getEtagForUpdate(id.toString());
+			transactionalMessenger.sendMessageAfterCommit(id.toString(), ObjectType.PRINCIPAL, etag, ChangeType.UPDATE);
+
 		}
 		
```

Comment:
String etag = userGroupDAO.getEtagForUpdate(groupId);
transactionalMessenger.sendMessageAfterCommit(id.toString(), ObjectType.PRINCIPAL, etag, ChangeType.UPDATE);

remove List<String> locks & for loop